R/stage_nwis_ts.R

```diff
@@ -113,13 +115,18 @@ stage_nwis_ts <- function(sites, var, times, folder = tempdir(), verbose = FALSE
     for (i in 1:length(un_sites)){
       
       site <- parse_site_name(un_sites[i])
-      site_data <- filter(nwis_data, site_no == site) %>%
-        select_('DateTime', var) %>%
+      site_data <- filter(nwis_data, site_no == site)
+      which.var <- select(site_data, -DateTime, -site_no) %>% 
+        tidyr::gather() %>% group_by(key) %>% summarize(keys = sum(!is.na(value))) %>% 
+        filter(keys==max(keys)) %>% select(key) %>% .$key
```

Comment:
this could break if two keys have an equal number of non-NA values.

also, i think you could replace `%>% select(key) %>% .$key` with just `%>% .$key`, right?

combining the two above thoughts, i think you could replace `%>% select(key) %>% .$key` with just `%>% .$key[1]`.